examples/3d/two_passes.rs

```diff
@@ -0,0 +1,203 @@
+
+    // Add a node for the first pass.
+    graph.add_node(FIRST_PASS_DRIVER, FirstPassCameraDriver);
+
+    // The first pass's dependencies include those of the main pass.
+    graph
+        .add_node_edge(node::MAIN_PASS_DEPENDENCIES, FIRST_PASS_DRIVER)
+        .unwrap();
+
+    // Insert the first pass node: CLEAR_PASS_DRIVER -> FIRST_PASS_DRIVER -> MAIN_PASS_DRIVER
+    graph
+        .add_node_edge(node::CLEAR_PASS_DRIVER, FIRST_PASS_DRIVER)
+        .unwrap();
+    graph
+        .add_node_edge(FIRST_PASS_DRIVER, node::MAIN_PASS_DRIVER)
+        .unwrap();
+    app.run();
+}
+
+// Add 3D render phases for FIRST_PASS_CAMERA.
+fn extract_first_pass_camera_phases(mut commands: Commands, active_cameras: Res<ActiveCameras>) {
+    if let Some(camera) = active_cameras.get(FIRST_PASS_CAMERA) {
+        if let Some(entity) = camera.entity {
+            commands.get_or_spawn(entity).insert_bundle((
+                RenderPhase::<Opaque3d>::default(),
+                RenderPhase::<AlphaMask3d>::default(),
+                RenderPhase::<Transparent3d>::default(),
+            ));
+        }
+    }
+}
+
+// A node for the first pass camera that runs draw_3d_graph with this camera.
+struct FirstPassCameraDriver;
+impl bevy::render::render_graph::Node for FirstPassCameraDriver {
+    fn run(
+        &self,
+        graph: &mut RenderGraphContext,
+        _render_context: &mut RenderContext,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        let extracted_cameras = world.get_resource::<ExtractedCameraNames>().unwrap();
+        if let Some(camera_3d) = extracted_cameras.entities.get(FIRST_PASS_CAMERA) {
+            graph.run_sub_graph(draw_3d_graph::NAME, vec![SlotValue::Entity(*camera_3d)])?;
+        }
+        Ok(())
+    }
+}
+
+// Marks the first pass cube.
+#[derive(Component)]
+struct FirstPassCube;
+
+// Marks the main pass cube.
+#[derive(Component)]
+struct MainPassCube;
+
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut active_cameras: ResMut<ActiveCameras>,
+) {
+    let cube_handle = meshes.add(Mesh::from(shape::Cube { size: 4.0 }));
+    let cube_material_handle = materials.add(StandardMaterial {
+        base_color: Color::GREEN,
+        reflectance: 0.02,
+        unlit: false,
+        ..Default::default()
+    });
+
+    let split = 2.0;
+
+    // This specifies the layer used for the first pass, which will be attached to the first pass camera and cube.
+    let first_pass_layer = RenderLayers::layer(1);
+
+    // The first pass cube.
+    commands
+        .spawn_bundle(PbrBundle {
+            mesh: cube_handle,
+            material: cube_material_handle,
+            transform: Transform::from_translation(Vec3::new(-split, 0.0, 1.0)),
+            ..Default::default()
+        })
+        .insert(FirstPassCube)
+        .insert(first_pass_layer);
```

Comment:
This is very similar to an API suggestion I made on discord, however it sounds like cart would prefer a slightly different direction: https://discord.com/channels/691052431525675048/931614194229477417/931974884534415380

In short, instead of adding a render layer component on rendered entities, the suggestion was that we should do the reverse - register the components we want to render in a pass, with that render pass. This leaves the open question of how to unregister entities from the main pass, because by default we want all 3d things to render into the main pass. The suggested answer was that we wrap `PbrBundle`s in a `Mesh3d` type, and register Mesh3d with the main pass. This way, if you want to render your mesh in another pass, you would make a `PbrBundle` and add your `MyUiMesh` component to it.

In cart's words:
> Gotcha. yeah feel free to experiment. I would also like to consider paths that don't need that extra abstraction. Ex: a Hud entity will be a "hud thing" by nature of having Hud components. A normal "3d mesh" entity will be a "main pass" thing by nature of having normal 3d mesh components.

Personally, I'm not convinced cart's suggestion as I understand it is a better route. I prefer something like what you've done here, in which case we might add a `MainPass` component to `PbrBundle` or a wrapper type - which is what we had in 0.5, the `MainPass` component was removed unintentionally.

One suggestion I'd like you to consider, regardless of the outcome of the above, is how plugins might interoperate. How could 3d HUD plugins compose if they all want to draw on top of the main pass? Is this something the user would be responsible for configuring manually? Could a render pass component be passed in as a plugin constructor argument?

Reply:
The API isn't great, but I'm mostly using the existing API (including `RenderLayers`, which I like.) I considered something like adding a Z-order to the `Camera`, but then composing cameras would be difficult, because you really want something more like the render graph, so you can express relations like "after the main pass, but before the UI". So I think it might be best to just rely on the render graph, and improve that API if needed e.g. avoid strings.

This iss